Adding Tls client information (#1344)
temporalio/documentation: docs/typescript/how-to-set-tls-configuration-in-typescript.md

---
id: how-to-set-tls-configuration-in-typescript
title: How to set TLS configuration in TypeScript
sidebar_label: Set TLS configuration
description: Set TLS configuration
tags:
  - developer-guide
  - sdk
  - typescript
---

To set the TLS configuration in TypeScript, use the [`tls`](https://typescript.temporal.io/api/interfaces/client.connectionoptions/#tls) connection option from the [`Client`](https://typescript.temporal.io/api/namespaces/client) class to connect to a Temporal Client with TLS.

```typescript
const connection = await Connection.connect({
  address: 'foo.bar.tmprl.cloud',
  tls: {
    clientCertPair: {
      crt: clientCertPath,
      key: clientKeyPath,
    },
  },
});
const client = new WorkflowClient({ connection, namespace: 'foo.bar' });
```

[The Hello World mTLS sample](https://github.com/temporalio/samples-typescript/tree/main/hello-world-mtls/) demonstrates sample code used to connect to a Temporal Cloud account.

> When signing up to Temporal Cloud you should receive a Namespace, a Server address and a Client certificate and key. Use the following environment variables to set up the sample:

Reviewer: ```suggestion
When signing up to Temporal Cloud, you should receive a Namespace, a Server address, and a Client certificate and key. Use the following environment variables to set up the sample:
```

- **TEMPORAL_ADDRESS**: looks like `foo.bar.tmprl.cloud` (NOT web.foo.bar.tmprl.cloud)
- **TEMPORAL_NAMESPACE**: looks like `foo.bar`
- **TEMPORAL_CLIENT_CERT_PATH**: `/tls/ca.pem` (file contents start with -----BEGIN CERTIFICATE-----)
- **TEMPORAL_CLIENT_KEY_PATH**: `/tls/ca.key` (file contents start with -----BEGIN PRIVATE KEY-----)

> You can leave the remaining vars, like `TEMPORAL_SERVER_NAME_OVERRIDE` and `TEMPORAL_SERVER_ROOT_CA_CERT_PATH` blank.
> There is another var, `TEMPORAL_TASK_QUEUE`, which the example defaults to `hello-world-mtls`, but you can customize as needed.
> Example environment settings

Reviewer: ```suggestion
You can leave the remaining variables, like `TEMPORAL_SERVER_NAME_OVERRIDE` and `TEMPORAL_SERVER_ROOT_CA_CERT_PATH`, blank.
If needed, you can customize `TEMPORAL_TASK_QUEUE`; the following example defaults to `hello-world-mtls`.
```

```typescript
export function getEnv(): Env {
  return {
    address: 'web.<Namespace_ID>.tmprl.cloud', // NOT web.foo.bar.tmprl.cloud
    namespace: 'your.namespace', // as assigned
    clientCertPath: 'foobar.pem', // in project root
    clientKeyPath: 'foobar.key', // in project root
    taskQueue: process.env.TEMPORAL_TASK_QUEUE || 'hello-world-mtls', // just to ensure task queue is same on client and worker, totally optional
    // // not usually needed
    // serverNameOverride: process.env.TEMPORAL_SERVER_NAME_OVERRIDE,
    // serverRootCACertificatePath: process.env.TEMPORAL_SERVER_ROOT_CA_CERT_PATH,
  };
}
```

> If you have misconfigured your connection somehow, you will get an opaque `[TransportError: transport error]` error. Read through your settings carefully and contact Temporal if you are sure you have checked everything.

Reviewer: ```suggestion
If you somehow misconfigure your connection, you get an opaque `[TransportError: transport error]` error.
Read through your settings carefully, and contact Temporal if you are sure you have checked everything.
```
